src/components/content/privacy-wrapper.tsx

```diff
@@ -0,0 +1,176 @@
+import {
+  faCubes,
+  faPlayCircle,
+  faSpinner,
+} from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import React, { ReactChild } from 'react'
+import styled from 'styled-components'
+
+import { instanceData } from '@/data/en'
+import { makeMargin, makePadding, makePrimaryButton } from '@/helper/css'
+
+// inspired by https://github.com/ibrahimcesar/react-lite-youtube-embed
+// also borrowed some code
+
+enum Provider {
+  YouTube = 'youtube',
+  WikimediaCommons = 'wikimedia',
+  Vimeo = 'vimeo',
+  GeoGebra = 'geogebra',
+}
+
+interface PrivacyWrapperProps {
+  children: ReactChild
+  placeholder?: ReactChild
+  type: 'video' | 'applet'
+  provider: Provider
+  previewImageUrl: string
+}
+
+function renderProvider(provider: Provider): string {
+  switch (provider) {
+    case Provider.YouTube:
+      return 'YouTube'
+    case Provider.WikimediaCommons:
+      return 'Wikimedia Commons'
+    case Provider.Vimeo:
+      return 'Vimeo'
+    case Provider.GeoGebra:
+      return 'GeoGebra'
+  }
+}
+
+export function PrivacyWrapper({
+  children,
+  placeholder,
+  type,
+  previewImageUrl,
+  provider,
+}: PrivacyWrapperProps) {
+  const [showIframe, setShowIframe] = React.useState(false)
+
+  const confirmLoad = () => {
+    if (showIframe) return
+    setShowIframe(true)
+  }
+
+  function onKeyDown(e: React.KeyboardEvent<HTMLButtonElement>) {
+    if (!showIframe && (e.key === 'Enter' || e.key === ' ')) {
+      e.preventDefault()
+      confirmLoad()
+    }
+  }
+
+  return (
+    <Wrapper>
+      {renderPlaceholder()}
+      {showIframe && children}
+    </Wrapper>
+  )
+
+  function renderPlaceholder() {
+    if (placeholder) return placeholder
+    const buttonLabel = instanceData.strings.embed[type]
+    const providerLabel = renderProvider(provider)
+
+    return (
+      <Placeholder>
+        {previewImageUrl && (
+          <PreviewImageWrapper>
+            <PreviewImage src={previewImageUrl} />
+          </PreviewImageWrapper>
+        )}
+        <InfoBar>
+          {instanceData.strings.embed.part1} &quot;{buttonLabel}&quot;
+          {instanceData.strings.embed.part2} {providerLabel}{' '}
+          {instanceData.strings.embed.part3} {providerLabel}{' '}
+          {instanceData.strings.embed.part4}{' '}
+          <a href="/privacy" target="_blank">
+            {instanceData.strings.embed.link}
+          </a>
+          .
```

Comment:
IMHO: we really should review the i18n stuff soon. Even this rather simple case was super annoying to add and might not even work with all languages (e.g. different grammar).

Reply:
Simplified it a bit. Please check if it still works for you.
We still need to work on the implementation of course :)